CMakeLists.txt

```diff
@@ -50,11 +50,11 @@ endif()
 
 # CXX flags setup
 if(NOT MSVC)
-    add_compile_options(-fPIC -Wall -Wpedantic
+    add_compile_options(-fPIC -Wall -Wpedantic -D_FORTIFY_SOURCE=2
```

Comment:
We could try adding `-D_FORTIFY_SOURCE=3` which adds additional runtime checks (supposedly with little to no performance impact).

In general, anything above 1 can break otherwise correct programs, so this might introduce a regression in SYCL...

Reply:
Should I add it under the UR_DEVELOPER_MODE flag?

Reply:
No, it's fine. This just might be a little bit more work than it appears since we don't have comprehensive regression testing :)

Reply:
For now let's change it to `-D_FORTIFY_SOURCE=1`, and we will revisit this once we have an e2e CI on UR end.

Reply:
For safety reasons, we should stay at level 2. I have added the required flags in this PR, but the PR may not build, as I am waiting for the @PatKamin PR with the flags wrapper.